scripts/serial-test.sh

```diff
@@ -1,36 +1,21 @@
 #!/bin/bash
-set -e # exit on failure of any "simple" command (excludes &&, ||, or | chains)
-# prepare environment
+set -eo pipefail
+[[ -f build.tar.gz ]] && tar -xzf build.tar.gz
+cd ./build
+echo "[Killing old MongoDB]"
+$(pgrep mongod | xargs kill -9) || true
+echo "[Starting MongoDB]"
 PATH=$PATH:~/opt/mongodb/bin
-echo "Extracting build directory..."
-[[ -z "${1}" ]] && tar -zxf build.tar.gz || tar -xzf $1
-echo "Starting MongoDB..."
-mongod --fork --dbpath ~/data/mongodb -f ~/etc/mongod.conf --logpath "$(pwd)"/mongod.log
-cd /data/job/build
-# run tests
-echo "Running tests..."
-TEST_COUNT=$(ctest -N -L nonparallelizable_tests | grep -i 'Total Tests: ' | cut -d ':' -f 2 | awk '{print $1}')
-[[ $TEST_COUNT > 0 ]] && echo "$TEST_COUNT tests found." || (echo "ERROR: No tests registered with ctest! Exiting..." && exit 1)
+mongod --fork --dbpath ~/data/mongodb -f ~/etc/mongod.conf --logpath ~/var/log/mongodb/mongod.log
+echo "[Running tests]"
 set +e # defer ctest error handling to end
-echo "$ ctest -L nonparallelizable_tests --output-on-failure -T Test"
 ctest -L nonparallelizable_tests --output-on-failure -T Test
 EXIT_STATUS=$?
-[[ "$EXIT_STATUS" == 0 ]] && set -e
-echo "Done running non-parallelizable tests."
-# upload artifacts
-echo "Uploading artifacts..."
-XML_FILENAME="test-results.xml"
-mv $(pwd)/Testing/$(ls $(pwd)/Testing/ | grep '20' | tail -n 1)/Test.xml $XML_FILENAME
-buildkite-agent artifact upload config.ini
-buildkite-agent artifact upload genesis.json
```

Comment:
As far as I know we don't use these artifacts, but someone put them there for a reason.  Are you sure they're not used?